yann/utils/image.py

```diff
@@ -45,7 +51,7 @@ def gray2rgb(r, g, b, channels_dim = 3):
     if channels_dim == 3:
         return out
     elif channels_dim == 1:
-        return out.transpose((0,3,1,2))
+        return out.transpose((2,0,1))
```

Comment:
Are you sure of this change ? This feels wrong to me. Can you verify this by creating a cifar 10 dataset and running it with pre_process rgb and then see how the visualizer prints down the data as images ?